### ISSUE TITLE
Version 7.0.0 does not work on iOS 18

### ISSUE THREAD

**Issue by Mullexoll** [csdcorp/speech_to_text#573]
Hello everyone. Please help to solve the problem. 
When updating to iOS 18, speech_to_text 7.0.0 does not work, shows the text but does not give a result. 
But on iOS below 18, everything works. When updating to 7.1.0-beta.3 on iOS 18, everything worked, but the same problem appeared on iOS 17.6.

` Future<void> _initSpeech() async {
    speechEnabled = await speechToText.initialize();
    setState(() {});
  }

  Future<void> _startListening() async {
    if (!isListening) {
      setState(() {
        isListening = true;
      });
      if (Platform.isIOS) await IOSSystemSound.playMicOnSound();

      final String currentLocale = GetIt.I<Locale>().languageCode;

      await speechToText.listen(
        pauseFor: Platform.isIOS ? const Duration(seconds: 2) : null,
        onResult: onSpeechResult,
        localeId: '${currentLocale}_${currentLocale.toUpperCase()}',
        listenOptions: SpeechListenOptions(
          listenMode: ListenMode.deviceDefault,
        ),
      );
      setState(() {});
    }
  }`
  There is no way to check for 17.6 now, because there is no real device, there are only devices for 18+.
Thanks for all))

**Comment by sowens-csd** [comment]
Sorry, not sure I understand correctly. Are you saying that beta.3 worked properly on iOS 18 but did not work on iOS 17.6?

**Comment by Mullexoll** [comment]
> Sorry, not sure I understand correctly. Are you saying that beta.3 worked properly on iOS 18 but did not work on iOS 17.6?

Yes, beta.3 is worked in ios 18 and not worked below 18, but v7.0.0 not worked on ios 18 and worked on below ios 18.

**Comment by Arpit144** [comment]
Any Update ?? I am facing. the same issue working on 17.7.2 or below but not working on 18.1.

**Comment by richanshah** [comment]
facing same issue any update on this?

**Comment by richanshah** [comment]
in 7.0.0 callback is not returning until manually stopped , pauseFor and listenFor seems to be not working properly
